app/controllers/profile_controller.rb

```diff
@@ -9,6 +13,35 @@ def show
     @user = current_user
   end
 
+  def password_form
+    @password_form = ChangePasswordForm.new({})
+  end
+
+  def change_password
+    @password_form = ChangePasswordForm.new(params[:change_password_form])
+    if @password_form.valid?
+      backend_change_password(
+        old: @password_form.old_password,
+        proposed: @password_form.password,
+        access_token: current_user.access_token
+      )
+      flash[:notice] = t('profile.password_changed')
+      redirect_to profile_path
+    else
+      flash.now[:errors] = @password_form.errors.full_messages.join(', ')
+      render :password_form, status: :bad_request
+    end
+  rescue InvalidOldPasswordError
+    flash[:error] = t('profile.old_password_mismatch')
+    render :password_form, status: :bad_request
+  rescue InvalidNewPasswordError
+    flash[:error] = t('devise.sessions.InvalidPasswordException')
```

Comment:
`t('devise.sessions.InvalidPasswordException')` this makes me 😢 

It's possible that Chris would like to have an error with the word `new` in the sentence. Might be worth having a specific error for that?